### DISCUSSION TITLE
Confusion about the best way to import Google Location history

### DISCUSSION THREAD

**Discussion by ant1fr** [Freika/dawarich#432] [Q&A]
Hi,  

I discovered Dawarich during my ongoing attempt to fully self-host everything related to personal data. Thanks for all the hard work! 🫶🏼  

Thanks to extensive Reddit monitoring, I managed to export my data from Google Takeout **before** switching to on-device location history.  
Currently, I have the following:  
- The raw `Records.json` location data  
- The Semantic Location History  
- The Google Takeout JSON file from my device (after switching to on-device timeline)  

Now, I’m a bit confused about the best way to proceed:  
1. What’s the best strategy for importing my data into Dawarich? Should I use only the `Records.json` file? Just the Semantic data? Or a combination of Semantic + `Records.json`? Or should I use the on-device export?  
2. Since 2009, I’ve used **many** different devices—sometimes multiple devices linked to my Google account simultaneously. When I tried importing the `Records.json` file, it generated aberrant locations that made it seem like I was teleporting between places at nearly the same time. As a result, the data in Dawarich is completely inconsistent. I’ve read a few threads about this issue, but I’m unsure about the best way to clean the raw data to ensure consistent locations and trips in Dawarich.  

Has anyone found an effective solution for this?

**Comment by Freika** [comment]
Hi @ant1fr and thanks for the kind words!

1. Import all of them, the order is not important. That way you'll have the most comprehensive picture of your location history. Dawarich validates points for lat-lon-timestamp uniqueness before creating, so no doubled points will be created anyway.
2. There is currently no solution to multiple devices tracking at the same time. However, if you'll be willing to send me your file and provide time period where two or more devices were tracking simultaneously, I could have a look and maybe start working on dividing the devices in the Dawarich interface so you could select what device data to render at the moment

**Comment by ant1fr** [comment]
Thanks!
I found a reference explaining the structure of the Google Takeout files and the way to isolate different devices in Records.json (see : https://locationhistoryformat.com/). But I can't find an easy way to determine which device should be considered as primary espacially as this can change over time.
Is there a way to send them to you privately?

**Reply by Freika** [reply]
Yup, there is an email address in my github profile: https://github.com/Freika/

**Reply by Freika** [reply]
btw the link you provided is _superuseful,_ thanks!

**Reply by STDog72** [reply]
Would be really nice if dawarich could import the location info (address, place names, etc) from the Semantic Location History vs external reverse geocoding.

**Comment by ovizii** [comment]
There seems to be another export option on Android:

> Go to device settings => location => location services => timeline export timeline data 

This leads to a timeline.json file. Not sure how this compares to the takeout mentioned above? Any ideas?

**Reply by Freika** [reply]
It should probably be imported with Google Phone Takeout as a source, can you try it?

**Reply by ovizii** [reply]
I gave it a try as Google Phone Takeout, I saw upload progress go up to 100%.
Then a message popped up, top, right corner, something along the lines of “Upload succeeded, ready to submit.” 

The upload progress is still showing 100%, there is a button:  “create import” just like before I started this. Lost in the process. If it succeeded, should the screen not hide the progress and possibly redirect me to the import page? 

I navigated to the imports page and it says:

> Hello there!
> Here you'll find your imports, but now there are none. Let's [create one](https://tline.foo.dj/imports/new)!

It would be really nice to be able to import this Teimeline.json as it's very easily to create it and its one file only. Maybe you can have a look at its format some time.

**EDIT: Scratch all that. It turns out, the upload started by itself and once done I needed to press the button “create import”.**

It is busy importing. I wonder if there is a way to check if the import worked 100%? I mean it looks like very few reverse geocoded points which begs the question, will this number increase later, were they originally not reverse geocoded or what does this number imply?

![image](https://github.com/user-attachments/assets/2910760b-104e-4d05-8ed2-827e5b93aa6f)

**Reply by ovizii** [reply]
It is done. Worked perfectly except that it created 103 notifications about “Error suggesting visits”. But let's not pollute this thread with that. Maybe you can add the info about this import method to your Wiki too.

> There seems to be another export option on Android:
> 
> Go to device settings => location => location services => timeline export timeline data
> 
> This leads to a timeline.json file.

![image](https://github.com/user-attachments/assets/f12482ce-b7ac-4495-a119-a4ef254ca965)

**Comment by bytegetter** [comment]
Bug:
Adding a File is only one time per filename possible. In the webinterface will nothing happen, only in the log you will find an entry. Is it possible to add an info in the webinterface or maybe renaming the uploaded file internally?

**Reply by bytegetter** [reply]
A file sometimes can only uplodet one time with the same filename. If the source was false selected (Records/Takeout), you can´t upload the same file until you rename it or restart dawarich.
